kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/MessageFetchContext.java

```diff
@@ -154,6 +155,37 @@ public void recycle() {
                             long offset = ((FetchRequest) fetchRequest.getRequest()).fetchData()
                                 .get(topicPartition).fetchOffset;
 
+                            // handle offset out-of-range exception
+                            ManagedLedgerImpl managedLedger = (ManagedLedgerImpl) tcm.getManagedLedger();
+                            try {
+                                long logStartOffset = MessageIdUtils.getLogStartOffset(managedLedger);
+                                long logEndOffset = MessageIdUtils.getLogEndOffset(managedLedger);
+                                if (offset > logEndOffset || offset < logStartOffset) {
+                                    log.error("Received request for offset {} for partition {}, "
+                                                    + "but we only have entries in the range {} to {}.",
+                                            offset, topicPartition, logStartOffset, logEndOffset);
+                                    responseData.put(topicPartition, new FetchResponse.PartitionData<>(
```

Comment:
If just retrurn an error response, how can kafka consumer fetch new messages?  
Assumes that the offset in Fetch request is less than LSO, can the consuemr or broker  move the offset to the LSO? If not, does the consume will block at the invilad offset?

Reply:
Kafka consumer will deal with out of range exception.